service/src/main/java/org/cbioportal/service/impl/GenericAssayServiceImpl.java

```diff
@@ -80,15 +80,15 @@
                 }
             } else {
                 List<GenericAssayAdditionalProperty> additionalProperties = genericAssayRepository.getGenericAssayAdditionalproperties(distinctStableIds);
+                Map<String, List<GenericAssayAdditionalProperty>> additionalPropertiesGroupedByStableId =
+                    additionalProperties.stream().collect(Collectors.groupingBy(GenericAssayAdditionalProperty::getStableId));
                 for (GenericAssayMeta meta : metaData) {
                     String stableId = meta.getStableId();
                     HashMap<String, String> map = new HashMap<>();
-                    List<GenericAssayAdditionalProperty> filteredAdditionalProperties = additionalProperties
-                            .stream()
-                            .filter(additionalProperty -> additionalProperty.getStableId().equals(stableId))
-                            .collect(Collectors.toList());
-                    for (GenericAssayAdditionalProperty additionalProperty : filteredAdditionalProperties) {
-                        map.put(additionalProperty.getName(), additionalProperty.getValue());
+                    if (additionalPropertiesGroupedByStableId.containsKey(stableId)) {
+                       for (GenericAssayAdditionalProperty additionalProperty : additionalPropertiesGroupedByStableId.get(stableId)) {
+                           map.put(additionalProperty.getName(), additionalProperty.getValue());
+                       }
                     }
                     meta.setGenericEntityMetaProperties(map);
```

Comment:
I'm not going to make you change this because I'd rather keep this pull request small, but in general, you should declare map types as `Map<`, rather than `HashMap<` even when the field is guaranteed to be initialized as a HashMap. If you aren't using any features specific to HashMap, this just makes your types a bit more honest about their requirements.

Reply:
Same goes for List / ArrayList, Set / HashSet